src/AppServices/DataSciencePage.tsx

```diff
@@ -1,55 +1,410 @@
+     */
+    [RHodsClusterAddonMode.Install]: [
+      <Button
+        onClick={() =>
+          console.log(
+            "Install action for cluster",
+            selectedCluster || clusters[0]
+          )
+        }
+        variant="primary"
+        key="install"
+      >
+        {t("installClusterAction")}
+      </Button>,
+      <Button
+        onClick={() => setIsModalOpen(false)}
+        key="cancel"
+        variant="secondary"
+      >
+        {t("cancelClusterAction")}
+      </Button>,
+    ],
+    /**
+     * How do I upgrade cluster for RHODS?
+     */
+    [RHodsClusterAddonMode.Upgrade]: [
+      <Button
+        onClick={() =>
+          console.log(
+            "Upgrade action for cluster",
+            upgradeStrategy,
+            selectedCluster || clusters[0]
+          )
+        }
+        variant="primary"
+        key="install"
+      >
+        {t(
+          upgradeStrategy === UpgradeStrategy.Create
+            ? "createClusterAction"
+            : "upgradeAction"
+        )}
+      </Button>,
+      <Button
+        onClick={() => setIsModalOpen(false)}
+        key="cancel"
+        variant="secondary"
+      >
+        {t("cancelClusterAction")}
+      </Button>,
+    ],
+    /**
+     * What is the link to Cluster manager create page?
+     */
+    [RHodsClusterAddonMode.Create]: [
+      <Button component="a" href="#" variant="primary" key="install">
+        {t("createClusterAction")}
+      </Button>,
+      <Button
+        onClick={() => setIsModalOpen(false)}
+        key="cancel"
+        variant="secondary"
+      >
+        {t("cancelClusterAction")}
+      </Button>,
+    ],
+  };
+
+  const getData = useCallback(async () => {
+    const { clusters, installableClusters } = await loadClusters();
+    if (installableClusters.length > 0) {
+      setClustersState({
+        mode: RHodsClusterAddonMode.Install,
+        clusters: installableClusters,
+      });
+    } else {
+      setClustersState({
+        mode:
+          clusters.length > 0
+            ? RHodsClusterAddonMode.Upgrade
+            : RHodsClusterAddonMode.Create,
+        clusters,
+      });
+    }
+  }, [loadClusters]);
+
+  const handleInstallModalOpen = () => {
+    // eslint-disable-next-line @typescript-eslint/no-floating-promises
+    getData();
+    setIsModalOpen(true);
+  };
+
+  const handleSelectCluster = (cluster: ClusterObject) => {
+    console.log(cluster);
+    setSelectedCluster(cluster);
+  };
 
-export const DataSciencePage: FunctionComponent = () => {
-  const { t } = useTranslation();
   return (
     <>
+      <Modal
+        isOpen={isModalOpen}
+        onClose={() => setIsModalOpen(false)}
+        variant="medium"
+        actions={modalActions[mode]}
+        title={t(clusterModalTitles[mode])}
+        description={
+          mode !== RHodsClusterAddonMode.Detecting ? (
+            <Title headingLevel="h2" size="md">
+              {t(clusterModalDescriptions[mode])}
```

Comment:
I think this will trigger an error with eslint. We disabled calls to `t` with dynamic strings. You could work around this by having the `clusterModalDescriptions` defined in the function itself, using `t` directly instead of just setting the translation key.